### PR TITLE
Add support for js in compose module

### PR DESCRIPTION
**Pull request by DRSchlaubi**

### CODE REVIEW

**Comment by mikepenz** [comment]
Thank you very much for the PR. 

We already have an experimental wasm based build available, which you can find here: https://github.com/mikepenz/AboutLibraries/releases/tag/v10.9.2-wasm1

**Comment by mikepenz** [comment]
With wasm now being available in more dependencies. wasm support was merged: https://github.com/mikepenz/AboutLibraries/pull/940

**Comment by DRSchlaubi** [comment]
I need a JS build however

**Comment by mikepenz** [comment]
@DRSchlaubi I am not sure how that would work, compose for js target without wasm. can you link me to the documentation explaining this?


The `core` module already has a `js` target which allows you to build any visualisation you would like: https://github.com/mikepenz/AboutLibraries/blob/develop/aboutlibraries-core/build.gradle.kts#L118

**Comment by DRSchlaubi** [comment]
It works exactly how I prred it, you just add the js target and add the canvas flag

could not find any specific documentation, except for the [wasm sample](https://github.com/Kotlin/kotlin-wasm-examples/blob/main/compose-imageviewer/webApp/build.gradle.kts#L25)

File: aboutlibraries-compose/gradle.properties [hunk 1]
```diff
@@ -1,3 +1,5 @@
 POM_NAME=AboutLibraries Library
 POM_DESCRIPTION=AboutLibraries automatically detects all dependencies of a project and collects their information including the license. Optionally visualising it via the provided ui components.
-POM_ARTIFACT_ID=aboutlibraries-compose
\ No newline at end of file
+POM_ARTIFACT_ID=aboutlibraries-compose
+
+org.jetbrains.compose.experimental.jscanvas.enabled=true
```
Comment (mikepenz) [inline aboutlibraries-compose/gradle.properties:5]: please revert this change as this is already part of the global gradle.properties

File: aboutlibraries-compose/build.gradle.kts [hunk 2]
```diff
@@ -69,14 +70,18 @@ compose {
 }
 
 kotlin {
-    targetHierarchy.default()
+    applyDefaultHierarchyTemplate()
```
Comment (mikepenz) [inline aboutlibraries-compose/build.gradle.kts:73]: please rebase, this change is already part of our default branch

File: aboutlibraries-compose/build.gradle.kts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import com.vanniktech.maven.publish.SonatypeHost
+import org.jetbrains.kotlin.gradle.targets.js.dsl.ExperimentalWasmDsl
```
Comment (mikepenz) [inline aboutlibraries-compose/build.gradle.kts:2]: please remove this added import

**Review by mikepenz — COMMENTED** [review]
Please rebase this PR on the latest develop branch. Happy to do a new release with `js` added if it just works like that.

**Comment by mikepenz** [comment]
Merged into dev (did a rebase for clean history, thus the auto close didn't function) 
https://github.com/mikepenz/AboutLibraries/commit/77469d905d690ebe7e9df6a4f0f932f162fff90a

**Comment by mikepenz** [comment]
In the process now to release 10.10.0-rc02

**Comment by mikepenz** [comment]
looks sonatype has some issues at this time: https://github.com/mikepenz/AboutLibraries/actions/runs/7252679539/job/19757656507#step:11:155
